pkg/providers/vsphere/vmprovider_vm_test.go

```diff
@@ -2825,6 +2826,285 @@ func vmTests() {
+						Expect(err).ToNot(HaveOccurred())
+						Expect(task.Wait(ctx)).To(Succeed())
+
+						// Create first snapshot CR
+						Expect(ctx.Client.Create(ctx, vmSnapshot)).To(Succeed())
+
+						// Create second snapshot
+						secondSnapshot = builder.DummyVirtualMachineSnapshot("", "test-second-snap", vm.Name)
+						secondSnapshot.Namespace = nsInfo.Namespace
+
+						task, err = vcVM.CreateSnapshot(ctx, secondSnapshot.Name, "second snapshot", false, false)
+						Expect(err).ToNot(HaveOccurred())
+						Expect(task.Wait(ctx)).To(Succeed())
+
+						// Create second snapshot CR
+						Expect(ctx.Client.Create(ctx, secondSnapshot)).To(Succeed())
+
+						// Set desired snapshot to first snapshot (revert from second to first)
+						vm.Spec.CurrentSnapshot = &vmopv1common.LocalObjectRef{
+							APIVersion: vmSnapshot.APIVersion,
+							Kind:       vmSnapshot.Kind,
+							Name:       vmSnapshot.Name,
+						}
+
+						err = createOrUpdateVM(ctx, vmProvider, vm)
+						Expect(err).ToNot(HaveOccurred())
+
+						// Verify VM status reflects the reverted snapshot
+						Expect(vm.Status.CurrentSnapshot).ToNot(BeNil())
+						Expect(vm.Status.CurrentSnapshot.Name).To(Equal(vmSnapshot.Name))
+
+						// Verify the snapshot is actually current in vCenter
+						var moVM mo.VirtualMachine
+						Expect(vcVM.Properties(ctx, vcVM.Reference(), []string{"snapshot"}, &moVM)).To(Succeed())
+						Expect(moVM.Snapshot).ToNot(BeNil())
+						Expect(moVM.Snapshot.CurrentSnapshot).ToNot(BeNil())
+
+						// Find the snapshot name in the tree to verify it matches
+						currentSnapName := vmlifecycle.FindSnapshotNameInTree(moVM.Snapshot.RootSnapshotList, moVM.Snapshot.CurrentSnapshot.Value)
+						Expect(currentSnapName).To(Equal(vmSnapshot.Name))
+					})
+				})
+
+				Context("when VM spec has nil CurrentSnapshot, but the VC VM has a snapshot", func() {
+					It("should not attempt revert and update status correctly", func() {
+
+						// Create VM with snapshot but don't set desired snapshot
+						var err error
+						vcVM, err = createOrUpdateAndGetVcVM(ctx, vmProvider, vm)
+						Expect(err).ToNot(HaveOccurred())
+
+						// Create snapshot in vCenter
+						task, err := vcVM.CreateSnapshot(ctx, vmSnapshot.Name, "test snapshot", false, false)
+						Expect(err).ToNot(HaveOccurred())
+						Expect(task.Wait(ctx)).To(Succeed())
+
+						// Create snapshot CR
+						Expect(ctx.Client.Create(ctx, vmSnapshot)).To(Succeed())
+
+						// Explicitly set CurrentSnapshot to nil
+						vm.Spec.CurrentSnapshot = nil
+
+						err = createOrUpdateVM(ctx, vmProvider, vm)
+						Expect(err).ToNot(HaveOccurred())
+
+						// Status should reflect the actual current snapshot
+						Expect(vm.Status.CurrentSnapshot).ToNot(BeNil())
+						Expect(vm.Status.CurrentSnapshot.Name).To(Equal(vmSnapshot.Name))
+					})
+				})
+			})
+
+			Context("Create/Update/Delete ISO backed VirtualMachine", func() {
```

Comment:
This is unrelated to this PR I assume

Reply:
Yes.  This got added from an incorrect merge after a rebase. :(
Fixed.